client/src/pages/test-suites/TestSuiteMainSection.tsx

```diff
@@ -1,153 +1,231 @@
 import * as React from "react";
-import { CiSaveDown2 } from "react-icons/ci";
 import { Rule, TestType, useTestStore } from "../../stores/testStore";
 import { useParams } from "react-router-dom";
 import { useSnackbar } from "../../hooks/useSnackbar";
-import { LoadingIcon } from "../sniffers/LoadingIcon";
-import { VscChecklist } from "react-icons/vsc";
-import { Tooltip } from "@mui/material";
 import { TestConfig } from "./TestConfig";
-import { TbAdjustmentsCog } from "react-icons/tb";
 import { ExecutionHistory } from "./ExecutionHistory";
+import TestsTopSection from "./TestsTopSection";
+import TabContext from "@mui/lab/TabContext";
+import RequestModal from "./RequestModal";
+import AssertionsModal from "./AssertionsModal";
 
 export const TestSuiteMainSection = () => {
```

Comment:
The functions here are repetitive with small changes between them. generally it confuses to read this code and understand the flow. I would this of a way to simplify it.